### PR TITLE
refactor(cdk/menu): use `inject` for all injection

### PR DESCRIPTION
**Pull request by jelbourn**
This change switches the cdk menu from using constructor injection to using the `inject` function. This was enabled by https://github.com/angular/angular/pull/45991. This approach has a number of advantages:

* We can add and remove injectables without changing the public constructor signature
* It reduces the boilerplate of passing arguments through `super`
* It avoids long/messy constructor blocks in favor of simpler member definitions.
* This sidesteps a common pitfall of making `@Optional` injection (`null` if not provided) as an optional parameter (`undefined` if not provided).

I expect this PR to involve some discussion around how we want to approach this in general.

### CODE REVIEW

File: src/cdk/menu/menu-trigger-base.ts [hunk 1]
```diff
@@ -36,6 +37,15 @@ export const MENU_TRIGGER = new InjectionToken<CdkMenuTriggerBase>('cdk-menu-tri
...
 })
 export abstract class CdkMenuTriggerBase implements OnDestroy {
+  /** The DI injector for this component. */
+  readonly injector = inject(Injector);
+
+  /** The view container ref for this component */
+  protected readonly viewContainerRef = inject(ViewContainerRef);
+
+  /** The menu stack in which this menu resides. */
+  protected readonly menuStack: MenuStack | null = inject(MENU_STACK, InjectFlags.Optional);
```
Comment (crisbeto) [inline src/cdk/menu/menu-trigger-base.ts:47]: Why is this optional now? It used to be required. It causes some extra null checks in the components extending this one.

Comment (jelbourn) [reply src/cdk/menu/menu-trigger-base.ts]: Bad copy and paste, removed the optional

**Comment by jelbourn** [comment]
CI failures are resolved. I had to disable the no bitwise operator lint since it's required for this API

**Comment by jelbourn** [comment]
Made another update to remove some unnecessary explicit types. e.g., if you have
```ts
class MenuTrigger {
  menuAim = inject(MENU_AIM, InjectFlags.Optional);
}
```
it will automatically infer `MenuAim | null` based on the `InjectionToken` type and the use of the second parameter to `inject`

**Comment by mmalerba** [comment]
Looks like still some lint & api failures

**Comment by jelbourn** [comment]
Whoops, I had it green and then messed up the formatting when I made that last edit. Fixed.

**Review by mmalerba — APPROVED** [review]

**Review by mmalerba — APPROVED** [review]

**Review by crisbeto — APPROVED** [review]
LGTM

**Pull request merged**